### PR TITLE
Device model

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
@johngian quick r?

File: feedthefox/devices/admin.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.contrib import admin
+from feedthefox.devices.models import Build, Device
```
Comment (johngian) [inline feedthefox/devices/admin.py:3]: Please add empty line between local `feedthefox` imports and `django`.

File: feedthefox/devices/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+from django.conf import settings
+import feedthefox.devices.models
```
Comment (johngian) [inline feedthefox/devices/migrations/0001_initial.py:7]: same here

File: feedthefox/devices/models.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+from django.utils.encoding import python_2_unicode_compatible
+
+
+def _get_upload_file_name(instance, filename):
+    os.path.join(settings.USER_PHOTOS_DIR, str(uuid.uuid4()) + '.jpg')
+
+
+@python_2_unicode_compatible
+class Build(models.Model):
+    """Model for FfxOS builds."""
```
Comment (johngian) [inline feedthefox/devices/models.py]: Is this a typo? Shouldn't it be `FxOS`?

File: feedthefox/devices/models.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    date = models.DateField(auto_now_add=True)
+    link = models.URLField(blank=True, default='')
+
+    def __str__(self):
+        return self.name
+
+
+@python_2_unicode_compatible
+class Device(models.Model):
+    """Model for FfxOS devices."""
```
Comment (johngian) [inline feedthefox/devices/models.py]: Same here

File: feedthefox/devices/models.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    model = models.CharField(max_length=120)
+    manufacturer = models.CharField(max_length=120)
+    imei = models.CharField(max_length=17, blank=True, default='')
+    user = models.ForeignKey(User, related_name='devices',
+                             blank=True, null=True, on_delete=models.SET_NULL)
+    image = models.ImageField(upload_to=_get_upload_file_name,
```
Comment (johngian) [inline feedthefox/devices/models.py]: I don't think that for `django >= 1.7` you need to define `upload_to`.

Comment (johngian) [inline feedthefox/devices/models.py]: https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.FileField.upload_to

File: feedthefox/settings.py [hunk 6]
```diff
@@ -98,6 +99,7 @@
 
 MEDIA_ROOT = config('MEDIA_ROOT', default=os.path.join(BASE_DIR, 'media'))
 MEDIA_URL = config('MEDIA_URL', '/media/')
+USER_PHOTOS_DIR = config('USER_PHOTOS_DIR', 'uploads/profiles')
```
Comment (johngian) [inline feedthefox/settings.py]: Why not use `media_url` directly? This is it's purpose anyway.

File: feedthefox/devices/models.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                              blank=True, null=True)
+    comments = models.TextField(blank=True, default='')
```
Comment (johngian) [inline feedthefox/devices/models.py]: Shouldn't this be `comment` since it's just a textfield?

**Comment by johngian** [comment]
r+wc

File: feedthefox/settings.py (hunk 6 above)
Comment (akatsoulas) [inline feedthefox/settings.py]: If we need more uploads in the future (eg device photos), it would be nice to have a structure in the media dir.

Comment (johngian) [inline feedthefox/settings.py]: It isn't a good idea not to use `/media` for deployment/sanity reasons. I get your point about structure. Please add this folder under /media eg:

`MEDIA_ROOT/uploads/profiles`

Comment (akatsoulas) [inline feedthefox/settings.py]: That was my intention! I just used it in .env and didn't append it as default option... Sorry about that!

File: feedthefox/devices/migrations/0001_initial.py (hunk 2 above)
Comment (akatsoulas) [inline feedthefox/devices/migrations/0001_initial.py:7]: I think we can skip migrations... :beer:

**Pull request merged**